gapis/api/vulkan/api/renderpass_framebuffer.api

```diff
@@ -340,6 +534,73 @@ sub void RecordSubpassBegin(ref!CommandBufferObject cb, u32 subpass) {
   }
 }
 
+sub ref!RenderPassBeginInfo createRenderPassBeginInfo(const VkRenderPassBeginInfo* pRenderPassBegin) {
+  info := pRenderPassBegin[0]
+  if !(info.renderPass in RenderPasses) { vkErrorInvalidRenderPass(info.renderPass) }
+  if !(info.framebuffer in Framebuffers) { vkErrorInvalidFramebuffer(info.framebuffer) }
+
+  beginInfo := new!RenderPassBeginInfo(
+    RenderPass: info.renderPass,
+    Framebuffer: info.framebuffer,
+    RenderArea: info.renderArea,
+  )
+
+  clearValues := info.pClearValues[0:info.clearValueCount]
+  for i in (0 .. info.clearValueCount) {
+    beginInfo.ClearValues[i] = clearValues[i]
+  }
+
+  // handle pNext
+  if info.pNext != null {
+    numPNext := numberOfPNext(info.pNext)
+    next := MutableVoidPtr(as!void*(info.pNext))
+    for i in (0 .. numPNext) {
+      sType := as!const VkStructureType*(next.Ptr)[0]
+      switch sType {
+        case VK_STRUCTURE_TYPE_DEVICE_GROUP_RENDER_PASS_BEGIN_INFO: {
+          ext := as!VkDeviceGroupRenderPassBeginInfo*(next.Ptr)[0]
+          beginInfo.DeviceGroupBeginInfo = new!DeviceGroupBeginInfo(
+            DeviceMask: ext.deviceMask,
+          )
+          rects := ext.pDeviceRenderAreas[0:ext.deviceRenderAreaCount]
+          for j in (0 .. ext.deviceRenderAreaCount) {
+            beginInfo.DeviceGroupBeginInfo.RenderAreas[j] = rects[j]
+          }
+        }
+      }
+      next.Ptr = as!VulkanStructHeader*(next.Ptr)[0].PNext
+    }
+  }
+
+  return beginInfo
+}
+
+sub ref!SubpassBeginInfo createSubpassBeginInfoFromContents(VkSubpassContents contents) {
+  return new!SubpassBeginInfo(
+    Contents: contents,
+  )
+}
+
+sub ref!SubpassBeginInfo createSubpassBeginInfoFromInfo(const VkSubpassBeginInfo* pSubpassBeginInfo) {
+  info := pSubpassBeginInfo[0]
+  // We do not support any extension for this functionality.
```

Comment:
Ensure `pNext` is null rather than ignoring it.

Reply:
Same comment with previous

Reply:
@pmuetschard

Reply:
Please check the pNext commit